metadata-ingestion/src/datahub/cli/docker_check.py

```diff
@@ -1,33 +1,93 @@
 import os
+import re
 from contextlib import contextmanager
 from typing import Iterator, List, Optional, Tuple
 
 import docker
 
-REQUIRED_CONTAINERS = [
-    "elasticsearch-setup",
-    "elasticsearch",
-    "datahub-gms",
-    "datahub-frontend-react",
-    "datahub-upgrade",
-    "schema-registry",
-    "broker",
-    "zookeeper",
-    # These two containers are not necessary - only helpful in debugging.
-    # "kafka-topics-ui",
-    # "schema-registry-ui",
-    # "kibana",
-    # "kafka-rest-proxy",
-    # "datahub-mce-consumer",
-    # "datahub-mae-consumer"
-]
+from datahub import __version__
+
+
+def version_tuple(version: Optional[str] = None) -> Tuple[int, ...]:
+    """
+    TODO: Probably need semantic version library here
+    """
+    if version not in ("master", "head"):
+        match = re.match(
+            "^v?([0-9]+[.][0-9]+([.][0-9]+){1,2})",
+            os.getenv("DATAHUB_VERSION", version if version else __version__),
+        )
+        if match and match.group(1) != "0.0.0":
+            tuple_version = tuple(map(int, (match.group(1).split("."))))
+            if len(tuple_version) == 3:
+                tuple_version = tuple_version + (0,)
+
+            return tuple_version
+
+    return 0, 0, 0, 0
+
+
+def required_containers(version: Optional[str] = None) -> List[str]:
+
+    comparable_version = version_tuple(version)
+
+    if comparable_version >= (0, 10, 0, 0) or comparable_version == (0, 0, 0, 0):
```

Comment:
would this mean that we add the compatibility here as if statements for future releases?

Reply:
No, from the yaml file configuration a conditional like logic would be written